src/bootstrap/bin/main.rs

```diff
@@ -7,78 +7,18 @@
 
 use std::env;
 
-use bootstrap::{Build, Config, Subcommand, VERSION};
+include!("../dylib_util.rs");
 
-fn main() {
-    let args = env::args().skip(1).collect::<Vec<_>>();
-    let config = Config::parse(&args);
-
-    // check_version warnings are not printed during setup
-    let changelog_suggestion =
-        if matches!(config.cmd, Subcommand::Setup { .. }) { None } else { check_version(&config) };
-
-    // NOTE: Since `./configure` generates a `config.toml`, distro maintainers will see the
-    // changelog warning, not the `x.py setup` message.
-    let suggest_setup = !config.config.exists() && !matches!(config.cmd, Subcommand::Setup { .. });
-    if suggest_setup {
-        println!("warning: you have not made a `config.toml`");
-        println!(
-            "help: consider running `./x.py setup` or copying `config.toml.example` by running \
-            `cp config.toml.example config.toml`"
-        );
-    } else if let Some(suggestion) = &changelog_suggestion {
-        println!("{}", suggestion);
-    }
+mod bootstrap;
+mod rustc;
+mod rustdoc;
 
-    let pre_commit = config.src.join(".git").join("hooks").join("pre-commit");
-    Build::new(config).build();
-
-    if suggest_setup {
-        println!("warning: you have not made a `config.toml`");
-        println!(
-            "help: consider running `./x.py setup` or copying `config.toml.example` by running \
-            `cp config.toml.example config.toml`"
-        );
-    } else if let Some(suggestion) = &changelog_suggestion {
-        println!("{}", suggestion);
-    }
-
-    // Give a warning if the pre-commit script is in pre-commit and not pre-push.
-    // HACK: Since the commit script uses hard links, we can't actually tell if it was installed by x.py setup or not.
-    // We could see if it's identical to src/etc/pre-push.sh, but pre-push may have been modified in the meantime.
-    // Instead, look for this comment, which is almost certainly not in any custom hook.
-    if std::fs::read_to_string(pre_commit).map_or(false, |contents| {
-        contents.contains("https://github.com/rust-lang/rust/issues/77620#issuecomment-705144570")
-    }) {
-        println!(
-            "warning: You have the pre-push script installed to .git/hooks/pre-commit. \
-                  Consider moving it to .git/hooks/pre-push instead, which runs less often."
-        );
-    }
-
-    if suggest_setup || changelog_suggestion.is_some() {
-        println!("note: this message was printed twice to make it more likely to be seen");
+fn main() {
+    match env::args().next().as_deref().and_then(|path| path.rsplit_once("/").map(|x| x.1)) {
```

Comment:
Does this handle Windows paths? I think a simpler alternative to rsplit_once is file_name().